event/event_target_bus.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+//go:build js
+
+package event
+
+import (
+	"sync"
+	"sync/atomic"
+	"syscall/js"
+
+	"github.com/sourcenetwork/goji"
+)
+
+type eventTargetSub struct {
+	id       uint64
+	listener js.Func
+	value    chan Message
+	events   []Name
+}
+
+// Message returns the message channel for the subscription.
+func (s *eventTargetSub) Message() <-chan Message {
+	return s.value
+}
+
+// eventTargetBus uses a JavaScript EventTarget to
+// manage subscribers and publish messages.
+type eventTargetBus struct {
```

Comment:
question: Did you consider decoupling this from JavaScript?  Could it perhaps be useful for non-browser hosted WASM builds, and other use-cases?